store/migration/10001__init_schema.sql

```diff
@@ -1395,10 +1395,28 @@ CREATE TABLE saved_query (
     created_ts BIGINT NOT NULL DEFAULT (strftime('%s', 'now')),
     updater_id INTEGER NOT NULL REFERENCES principal (id),
     updated_ts BIGINT NOT NULL DEFAULT (strftime('%s', 'now')),
+    project_id INTEGER NOT NULL REFERENCES project (id),
+    instance_id INTEGER NOT NULL REFERENCES instance (id),
+    database_id INTEGER NOT NULL REFERENCES db (id),
+    -- tab_id generate by front-end
+    tab_id TEXT NOT NULL UNIQUE,
     name TEXT NOT NULL,
-    statement TEXT NOT NULL
+    statement TEXT NOT NULL,
+    -- allowed access_type are 'PRIVATE', 'READ', 'WRITE'.
+    access_type TEXT NOT NULL DEFAULT '',
```

Comment:
I consider that the access_type values `PRIVATE` is for the owner, `READ` 、`WRITE` for team.

And the owner decides whether the SQL can view or write, maybe the other team members can write something on top of it. The owner can do some teamwork with other team members.

Reply:
'PRIVATE' and 'READ', 'WRITE' are disjoint properties and is not comparable. Otherwise it will be quite confusing.

e.g. One can create a private readonly note (note is locked to prevent accidental editing)


Another option is to combine them, like PRIVATE_RW, PRIVATE_RO, PROJECT_RO, PROJECT_RW, PUBLIC_RO, PUBLIC_RW.

Reply:
> 'PRIVATE' and 'READ', 'WRITE' are disjoint properties and is not comparable. Otherwise it will be quite confusing.
> 
> e.g. One can create a private readonly note (note is locked to prevent accidental editing)
> 
> Another option is to combine them, like PRIVATE_RW, PRIVATE_RO, PROJECT_RO, PROJECT_RW, PUBLIC_RO, PUBLIC_RW.

Maybe we can use two fields:

```sql
-- allowed access_type are 'READ', 'WRITE'.
access_type TEXT NOT NULL DEFAULT '',
-- allowed visibility are 'PRIVATE', 'TEAM', 'PUBLIC'.
visibility TEXT NOT NULL DEFAULT 'PRIVATE',
```
wdyt?

Reply:
Depending on how we treat the feature.

e.g. access_type = READ, visibility = TEAM.

Do we still allow owner to write or it's readonly for all?

I thought it's the former, then having a single field called access_level makes more sense since that suggests an order

PRIVATE_RO < PRIVATE_RW < PROJECT_RO <  PROJECT_RW <  PUBLIC_RO <  PUBLIC_RW

If we want to implement the readonly feature, we may add a separate locked column